File: mergeIntervals.js
const merge = (intervals) => {
    let previous = [null, -1]
    let result = [previous]
    intervals = intervals.sort((a, b) => a[0] - b[0])
    for (let i = 0; i < intervals.length; i++) {
        const current = intervals[i]
        if (current[0] > result[result.length - 1][1]) {
            result.push(current)
        } else {
            result[result.length-1][1] = Math.max(current[1], result[result.length-1][1])
        }
    }
    return result.slice(1, result.length)
}

File: mergeIntervals.py
def mergeIntervals(intervals):
    if len(intervals) == 0:
        return intervals
    intervals.sort(key = lambda x:x[0])
    result = [intervals[0]]
    for i in range(1, len(intervals)):
        current = intervals[i]
        previous = result[-1]
        if current[0] > previous[1]:
            result.append(current)
        else:
            result[-1][1] = max(previous[1], current[1])
    return result

print(mergeIntervals([[1,3],[2,6],[8,10],[15,18]]))
print(mergeIntervals([[1,4],[4,5]]))
